be/src/olap/rowset/segment_v2/binary_dict_page.cpp

```diff
@@ -152,7 +151,7 @@ size_t BinaryDictPageBuilder::count() const {
 }
 
 uint64_t BinaryDictPageBuilder::size() const {
-    return _pool.total_allocated_bytes() + _data_page_builder->size();
+    return _arena.size() + _data_page_builder->size();
```

Comment:
the same problem. The arena's allocated size maybe very large than the data's actual size. The segment maybe very small.

Reply:
same as above